### DISCUSSION TITLE
401 when attempting restore on Circle CI AWS Linux Machine

### DISCUSSION THREAD

**Discussion by JTCozart** [community/community#67032] [Packages]
### Select Topic Area

Question

### Body

I am using a private repository for the company I work with and I am trying to restore nuget packages on our build server (circle ci with an ubuntu agent) but when I attempt to restore the package I get a 401 from github. 

I have followed their instructions to a T but I am unable to get past the 401 (unauthorized) from their service. 

This is my config file for nuget. 
```html
<?xml version="1.0" encoding="utf-8"?>
<configuration>
    <packageSources>
        <clear />
        <add key="github" value="https://nuget.pkg.github.com/{{PRIVATE-REPO}}/index.json" />
    </packageSources>
    <packageSourceCredentials>
        <github>
            <add key="Username" value="JTCozart" />
            <add key="ClearTextPassword" value="{{TOKEN}}" />
        </github>
    </packageSourceCredentials>
</configuration>
``` 

When I attempt an dotnet restore I get the following. 
```
Determining projects to restore...
/usr/share/dotnet/sdk/7.0.401/NuGet.targets(158,5): warning : Your request could not be authenticated by the GitHub Packages service. Please ensure your access token is valid and has the appropriate scopes configured. [/home/circleci/project/{{PROJECT}/{{PROJECT}}.sln]
  Retrying 'FindPackagesByIdAsync' for source 'https://nuget.pkg.github.com/{{PRIVATE-REPO}}/download/{{NUGET-PACKAGE}}/index.json'.
  Response status code does not indicate success: 401 (Unauthorized).
```

Other users have suggested using the command to add nuget sources 
```
dotnet nuget add source "https://nuget.pkg.github.com/{{PRIVATE-REPO}}/index.json" --name "GitHub" --username USERNAME--password ACCESS_TOKEN
```
I have done this and get the same result. 

Other users have suggested disabling the public nuget repo and re-enabling it - This also does not work. 
I have verified time and time again that the token that I am using is correct. 
It feels like github has not implemented FindPackagesByIdAsync that dotnet cli is using to get the packages downloaded.

Kernel: 5.15.0-1039-aws 
Dotnet CLI version 7.0.401
NuGet Command Line: 6.7.0.127

**Comment by hafiz-muhammad** [comment]
1. Use a personal access token (classic) with the appropriate scope to authenticate with GitHub Packages. You can generate a personal access token by going to your GitHub account settings and selecting Developer settings > Personal access tokens. Make sure to select the appropriate scopes for the token, such as read:packages, write:packages, or delete:packages.

2. Check if the token you are using has the necessary permissions to access the private repository. Your personal account must have appropriate permissions to use or manage a package hosted by a package registry.

3. Use the dotnet nuget add source command to add the package source before running dotnet restore. You can use the GITHUB_TOKEN to access packages in the same repository or use the PAT token to access packages in another repository.

4. Make sure to replace the ClearTextPassword value in your nuget.config file with the PAT.

5. If you are still having issues, you may need to contact GitHub support for further assistance.

**Comment by JTCozart** [comment]
This is incredibly frustrating and I am not sure why this hasn't worked the first few times but I regenerated to token and it seems to have worked this time... Gonna close.
